src/coreclr/jit/valuenum.h

```diff
@@ -514,6 +514,14 @@ class ValueNumStore
 
     void PeelOffsets(ValueNum* vn, target_ssize_t* offset);
 
+    enum class PhiWalkResult
+    {
+        Continue,
+        Abort
+    };
+    typedef PhiWalkResult(WalkPhiVnsFn)(Compiler* comp, ValueNum vn);
+    bool VNWalkPhis(VNFuncApp phiDefFunc, WalkPhiVnsFn walkPhiVnsFn, int maxDepth = 2);
```

Comment:
It should rather be a template, similar to e.g. `FlowGraphNaturalLoop::VisitLoopBlocks`.

Reply:
Ah good point, it will allow to capture args and remove the Compiler* arg, addressed